test-engineering/contract-tests/client/tests/test_contile.py

```diff
@@ -3,34 +3,69 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 
-from typing import List
+from typing import Callable, Dict, List
 
 import pytest
 import requests
+from requests.models import Response
+
 from models import Step
 
 
-@pytest.fixture(name="contile_host")
-def fixture_contile_host(request):
-    """Read the contile host from the pytest config."""
+@pytest.fixture(name="hosts", scope="session")
+def fixture_hosts(request) -> Dict[str, str]:
+    """Return a dict mapping from a service name to a host name."""
+
+    return {
+        "contile": request.config.option.contile_url,
+        "partner": request.config.option.partner_url,
+    }
+
+
+@pytest.fixture(name="clear_partner_records")
+def fixture_clear_partner_records(hosts) -> Callable:
+    """Clear Contile request history on partner."""
+
+    partner_host = hosts["partner"]
+
+    def _clear_records():
+        r: Response = requests.request("DELETE", f"{partner_host}/records/")
+
+        error_message: str = (
+            f"Request status code was not as expected. Records may not have cleared "
+            f"after test run, details: '{r.text}'."
+        )
+
+        assert r.status_code == 204, error_message
+
+    return _clear_records
+
+
+@pytest.fixture(scope="function", autouse=True)
+def fixture_function_teardown(clear_partner_records: Callable):
+    """Execute instructions after each test."""
+
+    yield  # Allow test to execute
 
-    return request.config.option.contile_url
+    clear_partner_records()
 
 
-def test_contile(contile_host: str, steps: List[Step]):
+def test_contile(hosts: Dict[str, str], steps: List[Step]):
     """Test for requesting tiles from Contile."""
 
     for step in steps:
         # Each step in a test scenario consists of a request and a response.
         # Use the parameters to perform the request and verify the response.
 
-        method = step.request.method
-        url = f"{contile_host}{step.request.path}"
-        headers = {header.name: header.value for header in step.request.headers}
+        method: str = step.request.method
+        url: str = f"{hosts[step.request.service.value]}{step.request.path}"
+        headers: Dict[str, str] = {
+            header.name: header.value for header in step.request.headers
+        }
 
-        r = requests.request(method, url, headers=headers)
+        r: Response = requests.request(method, url, headers=headers)
```

Comment:
I'm wondering if we should import that as `RequestsResponse` or so to clearly distinguish that from our own local `Reponse` model. 🤔

Reply:
I agree that it would be good to distinguish the models, however would the right thing be to change the name of our `Response` model 😵‍💫? I'm thinking that `requests` definition of `Response` is the common understanding.

Reply:
We use `requests.models.Response` solely for type hints and I don't think we should design our types around a 3rd party dependency. Instead, I'd prefer if we bind these objects to a local name that makes sense for us OR use the namespace.

The `requests` library *exposes* the `Response` object directly, so we don't need to go through `requests.models`: https://github.com/psf/requests/blob/main/requests/__init__.py#L173

Bind to local name:

```python
from requests import Response as RequestsResponse
```

or, instead of:

```python
from requests import Response
r: Response = requests.request(method, url, headers=headers)
```

use the namespace:

```python
import requests
r: requests.Response = requests.request(method, url, headers=headers)
```

Reply:
Change applied: [529afcf](https://github.com/mozilla-services/contile/pull/422/commits/529afcf1b66575b0890b80c9498ab0aedcb6a112)

Going with the original suggestion of RequestsResponse.